packages/vuetify/src/components/VCarousel/VCarousel.tsx

```diff
@@ -105,6 +105,31 @@ export const VCarousel = genericComponent<new <T>(
       window.requestAnimationFrame(startTimeout)
     }
 
+    function onDelimiterKeyDown (e: KeyboardEvent, group: GroupProvide) {
+      if (props.direction === 'horizontal' && !['ArrowLeft', 'ArrowRight'].includes(e.key)) {
+        return
+      }
+
+      if (props.direction === 'vertical' && !['ArrowUp', 'ArrowDown'].includes(e.key)) {
+        return
+      }
+
+      e.preventDefault()
+
+      if (['ArrowLeft', 'ArrowUp'].includes(e.key)) {
+        group.prev()
+      }
+
+      if (['ArrowRight', 'ArrowDown'].includes(e.key)) {
+        group.next()
+      }
+
+      nextTick(() => {
+        const activeBtn = windowRef.value?.$el.querySelector('.v-btn--active')
+        activeBtn?.focus()
+      })
```

Comment:
Early returns and de-duplication would be fine if the method would not be named like a general purpose handler. If we can move them out (keydown handler in VWindow and this one) into separate place... (`utils/carousel.ts`?) it could be fine. Otherwise, the dumb if's might be more appropriate (same in VWindow.tsx)

```suggestion
      if (
        (props.direction === 'horizontal' && e.key === 'ArrowLeft') ||
        (props.direction === 'vertical' && e.key === 'ArrowUp')
      ) {
        e.preventDefault()
        group.prev()
        nextTick(() => windowRef.value?.$el.querySelector('.v-btn--active')?.focus())
      }

      if (
        (props.direction === 'horizontal' && e.key === 'ArrowRight') ||
        (props.direction === 'vertical' && e.key === 'ArrowDown')
      ) {
        e.preventDefault()
        group.next()
        nextTick(() => windowRef.value?.$el.querySelector('.v-btn--active')?.focus())
      }
```